contracts/contracts/levels/RecoveryFactory.sol

```diff
@@ -7,25 +7,26 @@ import './Recovery.sol';
 
 contract RecoveryFactory is Level {
 
-  address lostAddress;
+  mapping (address => address) lostAddress;
 
-  function createInstance(address) override public payable returns (address) {
+  function createInstance(address _player) override public payable returns (address) {
     Recovery recoveryInstance;
     recoveryInstance = new Recovery();
     // create a simple token 
     recoveryInstance.generateToken("InitialToken", uint(100000));
     // the lost address
-    lostAddress = address(uint160(uint256(keccak256(abi.encodePacked(uint8(0xd6), uint8(0x94), recoveryInstance, uint8(0x01))))));
+    lostAddress[_player] = address(uint160(uint256(keccak256(abi.encodePacked(uint8(0xd6), uint8(0x94), recoveryInstance, uint8(0x01))))));
```

Comment:
I think we should store it as `lostAddress[address(recoveryInstance)]`, because there could be many instances per player.

Reply:
this way, each new instance, overwrites the old one. Isn't this ideal ?